awx/api/versioning.py

```diff
@@ -24,6 +24,10 @@ def drf_reverse(viewname, args=None, kwargs=None, request=None, format=None, **e
     else:
         url = _reverse(viewname, args, kwargs, request, format, **extra)
 
+    if settings.OPTIONAL_API_URLPATTERN_PREFIX and request:
+        if request._request.path.startswith(f"/api/{settings.OPTIONAL_API_URLPATTERN_PREFIX}/v2/"):
```

Comment:
I'm kind of not sure what is intended here, but this almost certainly does not do whatever that is.  The condition virtually guarantees that the replace will be a no-op.